### PR TITLE
feat: validate store/add access

### PR DESCRIPTION
**Pull request by alanshaw**
This PR adds a request to w3access when a `store/add` invocation is handled.

The request to w3access ensures that the target space has been verified by a user via email.

### CODE REVIEW

**Review by vasco-santos — COMMENTED** [review]
Yes! I Like it

**Comment by olizilla** [comment]
@alanshaw plz to add PR description or link to issue or both!

File: api/service/store/add.js [hunk 1]
```diff
@@ -28,22 +28,25 @@ export function storeAddProvider(context) {
...
-      const account = capability.with
 
+      const resource = Server.DID.parse(capability.with).did()
       const [
+        verified,
         carIsLinkedToAccount,
         carExists
       ] = await Promise.all([
-        context.storeTable.exists(account, link.toString()),
+        context.access.verifyInvocation(invocation),
```
Comment (olizilla) [inline api/service/store/add.js:38]: do `context.access.verifyInvocation` before we do any other work? db and bucket reads have a smol $ cost.

File: api/functions/ucan-invocation-router.js [hunk 2]
```diff
@@ -35,7 +39,9 @@ async function ucanInvocationRouter (request) {
     }
   }
 
-  const server = await createUcantoServer({
+  const id = await getServiceDid()
```
Comment (olizilla) [inline api/functions/ucan-invocation-router.js:42]: plz long name like
```suggestion
  const uploadApiDid = await getServiceDid()
```

**Review by olizilla — APPROVED** [review]
👍

File: api/service/store/add.js (hunk 1 above)
Comment (alanshaw) [reply api/service/store/add.js:38]: IDK, I'm optimistically hoping that requests are going to have access to do this.

**Review by vasco-santos — COMMENTED** [review]
Can you extend this to `store/*`? and maybe `upload/*` if this lands after https://github.com/web3-storage/w3protocol/pull/184

File: api/test/utils.js [hunk 3]
```diff
@@ -77,3 +96,81 @@ export function getSigningOptions(ctx) {
     bucket: ctx.bucketName,
   }
 }
+
+/** @returns {Promise<MockAccess>} */
+export async function createAccessServer () {
```
Comment (vasco-santos) [inline api/test/utils.js:101]: Might be nice to consider creating a mock module for this as I expect it to be needed in more services. Not for now, just raising thoughts for visibility :)

**Review by vasco-santos — APPROVED** [review]
LGTM! 🎉  Thanks

**Comment by olizilla** [comment]
I had to make similar changes to createUcantoServer in #35 so let's get this one merged and I can deal with the conflicts over there.

**Pull request merged**